src/com/trilead/ssh2/crypto/digest/MAC.java

```diff
@@ -1,88 +1,147 @@
 
 package com.trilead.ssh2.crypto.digest;
 
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Mac;
+import javax.crypto.ShortBufferException;
+import javax.crypto.spec.SecretKeySpec;
+
 /**
  * MAC.
  * 
  * @author Christian Plattner, plattner@trilead.com
  * @version $Id: MAC.java,v 1.1 2007/10/15 12:49:57 cplattne Exp $
  */
-public final class MAC
+public final class MAC 
 {
-	Digest mac;
-	int size;
```

Comment:
These fields weren't private and may therefore have been referenced by some external callers by doing some funky stuff.  Deprecate them and mark them as Nullable is they're no longer used, but keep them around for binary compatibility for anyone upgrading between versions of this library.